src/hashing.rs

```diff
@@ -482,27 +482,30 @@ macro_rules! hash_domain {
     };
 }
 
-/// Creates a domain separated hasher.
+/// Creates a domain separated hasher type and domain in one
 #[macro_export]
-macro_rules! create_hasher {
-    ($digest:ty, $name:ident, $domain:expr, $label:expr, $version: expr) => {{
-        use $crate::hash_domain;
-
-        hash_domain!($name, $domain, $version);
-        let hasher = $crate::hashing::DomainSeparatedHasher::<$digest, $name>::new($label);
-        hasher
-    }};
-    ($digest: ty, $name:ident, $domain:expr, $label:expr) => {
-        create_hasher!($digest, $name, $domain, $label, 1)
+macro_rules! hasher {
+    ($digest:ty, $name:ident, $domain:expr, $version: expr, $mod_name:ident) => {
+        mod $mod_name {
+            use $crate::hash_domain;
+
+            hash_domain!(__HashDomain, $domain, $version);
+        }
+        pub type $name = $crate::hashing::DomainSeparatedHasher<$digest, $mod_name::__HashDomain>;
     };
-    ($digest: ty, $name:ident, $domain:expr) => {
-        create_hasher!($digest, $name, $domain, "", 1)
+    ($digest: ty, $name:ident, $domain:expr, $version: expr) => {
+        hasher!($digest, $name, $domain, $version, __inner);
```

Comment:
nit: 
Maybe use an even more specific default module name
```suggestion
        hasher!($digest, $name, $domain, $version, __inner_hasher_impl);
```

Reply:
cool, good idea